plugin/template_test.go

```diff
@@ -0,0 +1,201 @@
+package plugin
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/thriftrw/thriftrw-go/plugin/api"
+)
+
+func TestGoFileFromTemplate(t *testing.T) {
+	tests := []struct {
+		desc     string
+		template string
+		data     interface{}
+		options  []TemplateOption
+
+		wantBody  string
+		wantError string
+	}{
+		{
+			desc:     "simple",
+			template: "package foo",
+			wantBody: unlines("package foo"),
+		},
+		{
+			desc: "type reference",
+			template: `
+				package foo
+
+				var foo <formatType .> = nil
+			`,
+			data: &api.Type{
+				ReferenceType: &api.TypeReference{
+					Name:    "Foo",
+					Package: "github.com/thriftrw/thriftrw-go/bar",
+				},
+			},
+			wantBody: unlines(
+				`package foo`,
+				``,
+				`import "github.com/thriftrw/thriftrw-go/bar"`,
+				``,
+				`var foo bar.Foo = nil`,
+			),
+		},
+		{
+			desc: "type reference in the same package",
+			template: `
+				package bar
+
+				func hello() <formatType .> {
+					return nil
+				}
+			`,
+			data: &api.Type{
+				ReferenceType: &api.TypeReference{
+					Name:    "Foo",
+					Package: "github.com/thriftrw/thriftrw-go/hello",
```

Comment:
nit: here and below, can we make this package `github.com/thriftrw/thriftrw-go/bar` to match the package name in the template. it doesn't really matter but it's just less confusing that way